### PR TITLE
[FrameworkBundle] mitigate BC break with empty trusted_proxies

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 3.3
| Bug fix?      | kind of
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/symfony/symfony/pull/22238#issuecomment-305932238
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Comment by syrm** [comment]
One step closer to respect semver :-)

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -67,7 +67,11 @@ public function getConfigTreeBuilder()
                 ->end()
                 ->arrayNode('trusted_proxies') // @deprecated in version 3.3, to be removed in 4.0
                     ->beforeNormalization()
-                        ->always()
+                        ->ifTrue(function ($v) { return null === $v || array() === $v; })
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: if (empty($v)) ? would be aligned with if !empty below, isn't it?

Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: indeed

**Review by nicolas-grekas — APPROVED** [review]
:+1:

**Comment by sstok** [comment]
Looks good to me 👍 maybe add an exception-message expectation?

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**